contracts/TokenLocking.sol

```diff
@@ -0,0 +1,76 @@
+pragma solidity ^0.4.23;
+
+import "./ERC20Extended.sol";
+import "./IColonyNetwork.sol";
+import "./TokenLockingStorage.sol";
+import "../lib/dappsys/math.sol";
+
+contract TokenLocking is TokenLockingStorage, DSMath {
+  modifier tokenNotLocked(address _token) {
+    if (lockedBalances[_token][msg.sender] > 0) {
+      require(userTokenLockCount[_token][msg.sender] == totalTokenLockCount[_token]);
+    }
+    _;
+  }
+
+  modifier onlyColony() {
+    require(IColonyNetwork(colonyNetwork).isColony(msg.sender), "token-locking-sender-not-colony");
+    _;
+  }
+
+
+  function setColonyNetwork(address _colonyNetwork) public auth {
+    colonyNetwork = _colonyNetwork;
+  }
+
+  // Used for testing purposes
+  function getColonyNetwork() public view returns (address) {
+    return colonyNetwork;
+  }
+
+  function lockToken(address _token) public onlyColony {
+    totalTokenLockCount[_token] += 1;
+  }
+
+  function unlockTokenForUser(address _token, address _user) public onlyColony {
+    // This should never throw if you are using it right, so it can't be tested
+    assert(userTokenLockCount[_token][_user] < totalTokenLockCount[_token]);
+    userTokenLockCount[_token][_user] += 1;
+  }
+
+  function deposit(address _token, uint256 _amount) public
+  tokenNotLocked(_token)
+  {
+    // Update only if needed
+    if (userTokenLockCount[_token][msg.sender] < totalTokenLockCount[_token]) {
```

Comment:
Ok, thanks, i didn't know that :)

Reply:
Where has this logic gone?

Reply:
`userTokenLockCount < totalTokenLockCount` will never be `true`, since in `tokenNotLocked` modifier we require that `userTokenLockCount == totalTokenLockCount`